src/widgets/textinput.lua

```diff
@@ -4,28 +4,127 @@ function TextInput:initialize(options)
 	ikkuna.Widget.initialize(self)
 
 	self.focusable = true
+
+	self.editable = true
+	self.buffer = ''
+
+	self.cursorPosition = 0
+	self.cursorTime = 0
+	self.cursorVisible = false
+
+	self.masked = false
+	self.maskCharacter = '*'
+end
+
+function TextInput:update(delta)
+	ikkuna.Widget.update(self, delta)
+
+	self.cursorTime = self.cursorTime + delta
+	if self.cursorTime > 0.35 then
+		self.cursorVisible = not self.cursorVisible
+		self.cursorTime = 0
+	end
+end
+
+function TextInput:draw()
+	ikkuna.Widget.draw(self)
+
+	if self.cursorVisible then
+		local width = 0
+		if #self.buffer > 0 and self.cursorPosition > 0 then
+			width = ikkuna.font:getWidth(self.buffer:sub(1, self.cursorPosition))
+		end
+
+		local height = ikkuna.font:getHeight()
+
+		love.graphics.setColor(1, 1, 1)
+		love.graphics.line(self.x + width, self.y, self.x + width, self.y + height)
+	end
 end
 
 function TextInput:onMousePressed(x, y, button, touch, presses)
 	return true
 end
 
 function TextInput:onTextInput(text)
+	if not self.editable then
+		return true
```

Comment:
Shouldn't this return `false`? If not, then shouldn't the `code` check also return true?